File: Everything/ClientApp/src/models/todo/ToDoColumn.ts
import ToDoItem from "./ToDoItem";

export default class ToDoColumn {
    id: number = 0;
    toDoBoardId: number = 0;
    name: string | null = null;
    createdDate: Date | null = null;
    description: string = '';
    sequence: number = 0;
    toDoItems: ToDoItem[] = [];

    constructor(init?: Partial<ToDoColumn>) {
        Object.assign(this, init);
    }
}

File: Everything/Models/ToDoList/ToDoColumn.cs
using System;
using System.Collections.Generic;

namespace everything.Models
{
    public class ToDoColumn
    {
        public ToDoColumn()
        {
            ToDoItems = new List<ToDoItem>();
        }

        public int Id { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Sequence { get; set; }

        public int ToDoBoardId { get; set; }
        public virtual ToDoBoard ToDoBoard { get; set; }

        public virtual ICollection<ToDoItem> ToDoItems { get; set; }
    }
}
